hack/tests/e2e-ansible-molecule.sh

```diff
@@ -11,7 +11,8 @@ ROOTDIR="$(pwd)"
 TMPDIR="$(mktemp -d)"
 trap_add 'rm -rf $TMPDIR' EXIT
 pip3 install --user pyasn1==0.4.7 pyasn1-modules==0.2.6 idna==2.8 ipaddress==1.0.22
-pip3 install --user molecule==2.22
+pip3 install --user molecule==3.0.2
```

Comment:
```suggestion
pip3 install --user molecule~=3.0
```
Not sure how you feel about this but it probably makes sense to pin to a minor version so we don't lag too far behind if an issue does pop up (which it shouldn't)

Reply:
The problem is that it is used in the tests. So, we could start to get an upper version and face some issue just because of this. So, in order to ensure that the tests will validate only the changes performed, I think we should use a specific version.

Reply:
But in the docs I think is fine we say that 👍